misc/python/materialize/cli/run.py

```diff
@@ -278,7 +287,16 @@ def main() -> int:
             ]
             env["MZ_SYSTEM_PARAMETER_DEFAULT"] = ";".join(formatted_params)
             db = urlparse(args.postgres).path.removeprefix("/")
-            _run_sql(args.postgres, f"CREATE DATABASE IF NOT EXISTS {db}")
+            if db:
+                url_without_db = "/".join(args.postgres.split("/")[:-1])
+                if (
+                    _capture_sql(
+                        url_without_db,
+                        f"SELECT 1 FROM pg_database WHERE datname='{db}'",
+                    )
+                    != "1"
+                ):
+                    _run_sql(url_without_db, f"CREATE DATABASE {db}")
```

Comment:
Sorry—I still don't follow the intent of these logic changes. A few folks are having trouble with the new logic here, so I'm going to take a stab at simplifying this.

Reply:
I can also revert the change in this file entirely, it was only for my local testing where I didn't specify a database explicitly.